### PR TITLE
[stable/external-dns] fixes kubernetes-sigs/external-dns#1262 AWS ARN issues

### PR DESCRIPTION
**Pull request by Freyert**
TLDR;
unnecessary source_profile
breaks config
in new aws sdk

#### Is this a new chart
No

#### What this PR does / why we need it:
More recent versions of the AWS Go SDK throw errors if your AWS config has `source_profile`, and does not have any of the other necessary fields specified in their documentation here: https://godoc.org/github.com/aws/aws-sdk-go/aws/session#hdr-Assume_Role_configuration

AWS documentation isn't really clear on how credentials are mapped from `config` to `credentials`, but I think given the examples [here](https://docs.aws.amazon.com/cli/latest/userguide//cli-configure-files.html) that it maps the profile name to the credential name.

Reading the doc for `source_profile` seems to confirm this. The doc mentions that it points to a `profile` which is a thing that only exists in `config`. Items in `credentials` are credentials.

Have also tested this locally and the pod does run.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - https://github.com/kubernetes-sigs/external-dns/issues/1262

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Review by sameersbn — APPROVED** [review]
/ok-to-test
/lgtm
